### PR TITLE
IGNITE-23948 Add print size only for MultipleRowReplicaRequest#binaryTuples in IgniteToStringBuilder#toString

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-23948

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/tostring/IgniteToStringBuilder.java [hunk 1]
```diff
@@ -894,7 +893,7 @@ private static void toString(StringBuilderLimitedLength buf, Object val) {
      * @param cls value class.
      * @param val value to print.
      */
-    private static void toString(StringBuilderLimitedLength buf, Class<?> cls, Object val) {
+    private static void toString(StringBuilderLimitedLength buf, Class<?> cls, @Nullable Object val) {
```
Comment (Phillippko) [inline modules/core/src/main/java/org/apache/ignite/internal/tostring/IgniteToStringBuilder.java:896]: cls is also Nullable

**Review by Phillippko — APPROVED** [review]

File: modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/messages/MessageImplGenerator.java [hunk 2]
```diff
@@ -137,12 +139,20 @@ public TypeSpec generateMessageImpl(MessageClass message, TypeSpec builderInterf
...
                 IgniteToStringInclude includeAnnotation = getter.getAnnotation(IgniteToStringInclude.class);
-
-                AnnotationSpec includeAnnotationSpec = includeAnnotation == null
-                        ? AnnotationSpec.builder(IgniteToStringInclude.class).build()
-                        : AnnotationSpec.get(includeAnnotation);
-
-                fieldBuilder.addAnnotation(includeAnnotationSpec);
+                IgniteStringifier stringifierAnnotation = getter.getAnnotation(IgniteStringifier.class);
+
+                if (stringifierAnnotation != null) {
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/messages/MessageImplGenerator.java:144]: This is how it is intended, the user himself will create the string for output, without automatic hiding of sensitive data.

[the remark this answers is not in the document]
Comment (Phillippko) [reply modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/messages/MessageImplGenerator.java:144]: Inconsistent behaviour, in **IgniteToStringBuilder** we don't ignore this field: 

```if (incFld != null || incType != null) {
                // Information is not sensitive when both the field and the field type are not sensitive.
                // When @IgniteToStringInclude is not present then the flag is false by default for that attribute.
                boolean notSens = (incFld == null || !incFld.sensitive()) && (incType == null || !incType.sensitive());
                add = notSens || includeSensitive();
            } else if (stringifierField != null || stringifierType != null) {
                add = true;
```

Could we forbid setting both **IgniteToStringInclude** and **IgniteStringifier**? And **IgniteToStringInclude** / **IgniteStringifier** ideally should be mutually exclusive with **IgniteToStringExlude**

[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/messages/MessageImplGenerator.java:144]: Fix the code in **IgniteToStringBuilder** as discussed in a private messages.

**Pull request merged**